test/suites/peeked-suite.ts

```diff
@@ -0,0 +1,76 @@
+import test = require('tape');
+import {ReadableSignal, Signal} from '../../src';
+import {LeakDetectionSignal} from '../lib/leak-detection-signal';
+import {parentChildSuite} from './parent-child-suite';
+
+export type PeekedSignalCreationFunction = <T>(
+    baseSignal: ReadableSignal<T>,
+    peekaboo: (payload: T) => void,
+) => ReadableSignal<T>;
+export function peekedSuite(prefix: string, createPeekedSignal: PeekedSignalCreationFunction) {
+    parentChildSuite(prefix, () => {
+        const parentSignal = new Signal();
+        const childSignal = createPeekedSignal(parentSignal, _payload => void 0);
+        return { parentSignal, childSignal };
+    });
+
+    test(`${prefix} should be called but not modify the payload`, t => {
+        const baseSignal = new Signal<number>();
+
+        let sideEffectCatcher!: number;
+        let actualReceiver!: number;
+        baseSignal
+            .map(x => x * 3)
+            .peek(x => { sideEffectCatcher = x; })
+            .add(x => { actualReceiver = x; });
+
+        baseSignal.dispatch(4);
+
+        t.deepEqual(sideEffectCatcher, 12);
+        t.deepEqual(actualReceiver, 12);
+
+        baseSignal.dispatch(7);
+
+        t.deepEqual(sideEffectCatcher, 21);
+        t.deepEqual(actualReceiver, 21);
+
+        t.end();
+    });
+
+    test(`${prefix} should be called but not modify the payload`, t => {
+        const baseSignal = new Signal<number>();
+
+        const sideEffectCatcher1: number[] = [];
+        const sideEffectCatcher2: number[] = [];
+        const actualReceiver: number[] = [];
+
+        baseSignal
+            .peek(x => { sideEffectCatcher1.push(x); })
+            .map(x => x * 3)
+            .peek(x => { sideEffectCatcher2.push(x); })
+            .filter(x => x % 2 === 0)
+            .add(x => { actualReceiver.push(x); });
+
+        [1, 5, 6, 7, 12].forEach(x => baseSignal.dispatch(x));
+
+        t.deepEqual(sideEffectCatcher1, [1, 5, 6, 7, 12]);
+        t.deepEqual(sideEffectCatcher2, [3, 15, 18, 21, 36]);
+
+        t.deepEqual(actualReceiver, [18, 36]);
+
+        t.end();
+    });
+
+    test('ReducedSignal should not leak', t => {
```

Comment:
ReducedSignal -> PeekedSignal?

Reply:
oh no, you discovered that I copy'n'pasted your tests O_O